app/src/main/java/com/projectfkklp/saristorepos/activities/dashboard/DashboardPage.java

```diff
@@ -54,6 +71,80 @@ private void generateAnalyticsChart(){
+
+        // Set the data for the PieChart
+        todaySalesChart.salesSetData(todaySalesEntries, new ValueFormatter() {
+            @Override
+            public String getFormattedValue(float value) {
+                return "";
+            }
+        });
+
+        // Generate center text with customizable parameters
+        String centerText = generateCenterText(overallSalesPercentage, actualSales, overallSalesTarget, incomeFromYesterday);
+
+        // Set center text for the PieChart
+        todaySalesChart.setCenterText(getSpannableString(centerText));
+        todaySalesChart.setCenterTextSize(8f);
+
+        // Move the center text to the center of the hole
+        todaySalesChart.setCenterTextOffset(0f, 0f);
+    }
+
+    // Method to generate customizable center text
+    private String generateCenterText(float overallSalesPercentage, int actualSales, int overallSalesTarget, int incomeFromYesterday) {
+        // Change the color of specific text to green
+        String centerText = String.format("Today Sales\n%.0f%% to Goal\nActual Sales: %s\nTarget Sales: %s\n\uD83D\uDCC8%d%% vs Yesterday",
+                overallSalesPercentage,
+                StringUtils.formatToPeso((float) actualSales),
+                StringUtils.formatToPeso((float) (overallSalesTarget - actualSales)),
+                incomeFromYesterday);
+
+        // Modify the color of "Today Sales" and "10%"
+        centerText = centerText.replace("Today Sales", "Today Sales ")
+                .replace("10%", "10%").replace("64%", "64%");
+        return centerText;
+    }
+
+    // Method to apply SpannableString for text styling
+    private SpannableString getSpannableString(String text) {
+        SpannableString spannableString = new SpannableString(text);
+
+        // Change the color of "Today Sales" and "10%" to green
+        int blueColor = Color.parseColor("#0000FF");
+        int greenColor = Color.parseColor("#40C94F");
+        int crimsonColor = Color.parseColor("#DC143C"); // This is the hex color code for Crimson
+
+
+        int startIndex = text.indexOf("Today Sales");
+        int endIndex = startIndex + 11; // Length of "Today Sales"
+        spannableString.setSpan(new ForegroundColorSpan(blueColor), startIndex, endIndex, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);
+        spannableString.setSpan(new StyleSpan(Typeface.BOLD), startIndex, endIndex, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);
+
+        startIndex = text.indexOf("10%");
+        endIndex = startIndex + 3; // Length of "10%"
+        spannableString.setSpan(new ForegroundColorSpan(greenColor), startIndex, endIndex, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);
+
+        startIndex = text.indexOf("64%");
+        endIndex = startIndex + 3; // Length of "64%"
+        spannableString.setSpan(new ForegroundColorSpan(crimsonColor), startIndex, endIndex, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);
+
+        return spannableString;
+    }
+
+
```

Comment:
These functions should be inside of DashboardTodaySalesChart class (you need to create new class to isolate it from TopCharts)